packages/grid/src/DataBarCellRenderer.ts

```diff
@@ -0,0 +1,614 @@
+          const markerValue = Number(model.textForCell(markerColumn, modelRow));
+          return markerValue <= 0
+            ? zeroPosition +
+                (Math.abs(markerValue) / columnLongest) * (maxWidth / 2)
+            : zeroPosition -
+                (Math.abs(markerValue) / columnLongest) * (maxWidth / 2);
+        });
+      }
+    } else if (axis === 'directional') {
+      const columnLongest = Math.max(Math.abs(columnMin), Math.abs(columnMax));
+      if (direction === 'LTR') {
+        zeroPosition = 0;
+        dataBarX = zeroPosition;
+        markerXs = markers.map(marker => {
+          const { column: markerColumn } = marker;
+          const markerValue = Number(model.textForCell(markerColumn, modelRow));
+          return (
+            zeroPosition + (Math.abs(markerValue) / columnLongest) * maxWidth
+          );
+        });
+      } else if (direction === 'RTL') {
+        zeroPosition = columnWidth;
+        dataBarX = zeroPosition - (Math.abs(value) / columnLongest) * maxWidth;
+        markerXs = markers.map(marker => {
+          const { column: markerColumn } = marker;
+          const markerValue = Number(model.textForCell(markerColumn, modelRow));
+          return (
+            zeroPosition - (Math.abs(markerValue) / columnLongest) * maxWidth
+          );
+        });
+      }
+    }
+
+    if (direction === 'LTR') {
+      zeroPosition += x + leftPadding;
+      dataBarX += x + leftPadding;
+      markerXs = markerXs.map(markerX => markerX + x + leftPadding);
+
+      if (valuePlacement === 'beside' && textAlign === 'left') {
+        zeroPosition += longestValueWidth + cellHorizontalPadding;
+        dataBarX += longestValueWidth + cellHorizontalPadding;
+        markerXs = markerXs.map(
+          markerX => markerX + longestValueWidth + cellHorizontalPadding
+        );
+      }
+    } else if (direction === 'RTL') {
+      zeroPosition = zeroPosition + x - rightPadding;
+      dataBarX = dataBarX + x - rightPadding;
+      markerXs = markerXs.map(markerX => markerX + x - rightPadding);
+
+      if (valuePlacement === 'beside' && textAlign === 'right') {
+        zeroPosition = zeroPosition - cellHorizontalPadding - longestValueWidth;
+        dataBarX = dataBarX - cellHorizontalPadding - longestValueWidth;
+        markerXs = markerXs.map(
+          markerX => markerX - cellHorizontalPadding - longestValueWidth
+        );
+      }
+    }
+
+    leftmostPosition += x;
+    rightmostPosition += x;
+
+    return {
+      width: maxWidth,
+      x: dataBarX,
+      y: y + 1,
+      zeroPosition,
+      leftmostPosition,
+      rightmostPosition,
+      markerXs,
+    };
+  }
+
+  drawGradient(
+    context: CanvasRenderingContext2D,
+    leftColor: Oklab,
+    rightColor: Oklab,
+    x: number,
+    y: number,
+    width: number,
+    height: number
+  ) {
+    let currentColor = leftColor;
+    for (let currentX = x; currentX <= x + width; currentX += 1) {
```

Comment:
I think the issue we've observed with the black lines in the gradient at certain zooms is some sort of interpolation effect when drawn at half pixels... or something.
Anyway, I think it's fixed if you just advance by half a pixel instead of a full pixel:
```suggestion
    for (let currentX = x; currentX <= x + width; currentX += 0.5) {
```
Not sure if there's another underlying cause for it. But that seems to resolve it.